### ISSUE TITLE
Support multiple filters in event source mappings

### ISSUE THREAD

**Issue by moritzzimmer** [terraform-aws-modules/terraform-aws-lambda#350]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality

## Is your request related to a problem? 

[Lambda supports](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-syntax) up to five different filters for a single event source. It seems like the module currently only supports defining one [filter](https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/8f99aaaf1af1155d71b49a377727af480af03df6/main.tf#L295).

```hcl
dynamic "filter_criteria" {
    for_each = try(each.value.filter_criteria, null) != null ? [true] : []

    content {
      filter {
        pattern = try(each.value["filter_criteria"].pattern, null)
      }
    }
  }
```
## Describe the solution you'd like.

Support multiple `filter` elements in `filter_criteria` (e.g. by providing a `list` of filters) in the `aws_lambda_event_source_mapping`:

```hcl
dynamic "filter_criteria" {
    for_each = lookup(each.value, "filter_criteria", null) != null ? [true] : []

    content {
      dynamic "filter" {
        for_each = try(each.value.filter_criteria, [])
        content {
          pattern = try(filter.value.pattern, "")
        }
      }
    }
  }
```

example usage:

```hcl
filter_criteria = [
        {
          pattern = jsonencode({
            data : {
              foo : ["bar"]
            }
          })
        },
        {
          pattern = jsonencode({
            data : {
              bar : ["foo"]
            }
          })
        }
      ]
    }
```

**Comment by antonbabenko** [comment]
PR is welcome, as always

**Comment by moritzzimmer** [comment]
I'll work on a PR for this

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.10.0 :tada:
